client/src/utils/lmcSimulator.js

```
@@ -68,6 +97,10 @@ class LMC {
     return this.programIndices;
   }
 
+  addToProgramIndices(index) {
+    return (this.programIndices = [...this.programIndices, index]);
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)